cancer_ai/rewarder_test.py

```diff
@@ -0,0 +1,301 @@
+
+        self.assertAlmostEqual(rewarder.scores["leader-1"], expected_share_leader_1, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-2"], expected_share_leader_2, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-3"], expected_share_leader_3, places=2)
+
+    def test_three_competitions_three_leaders_all_different_reductions(self):
+        """Test case 5: All competitors have different degrees of reduced shares"""
+        competitions_leaders = {
+            "competition-1": CompetitionLeader(hotkey="leader-1", leader_since=datetime.now() - timedelta(days=21)),
+            "competition-2": CompetitionLeader(hotkey="leader-2", leader_since=datetime.now() - timedelta(days=35)),
+            "competition-3": CompetitionLeader(hotkey="leader-3", leader_since=datetime.now() - timedelta(days=49))
+        }
+        scores = {"leader-1": 0, "leader-2": 0, "leader-3": 0}
+        rewarder_config = RewarderConfig(competition_leader_mapping=competitions_leaders, scores=scores)
+        rewarder = Rewarder(config=rewarder_config)
+
+        rewarder.update_scores()
+
+        base_share = 1 / 3
+        reduction_factor_leader_1 = 0.9  # 10% reduction
+        reduction_factor_leader_2 = 0.7  # 30% reduction
+        reduction_factor_leader_3 = 0.5  # 50% reduction
+
+        expected_share_leader_1 = base_share * reduction_factor_leader_1
+        expected_share_leader_2 = base_share * reduction_factor_leader_2
+        expected_share_leader_3 = base_share * reduction_factor_leader_3
+        # Calculate distributed reduction
+        remaining_share_leader_1 = base_share - expected_share_leader_1
+        remaining_share_leader_2 = base_share - expected_share_leader_2
+        remaining_share_leader_3 = base_share - expected_share_leader_3
+
+        # Leaders 2 and 3 gets their base share plus the distributed reduction from Leader 1
+        expected_share_leader_2 += remaining_share_leader_1 / 2
+        expected_share_leader_3 += remaining_share_leader_1 / 2
+
+        # Leaders 1 and 3 gets their base share plus the distributed reduction from Leader 2
+        expected_share_leader_1 += remaining_share_leader_2 / 2
+        expected_share_leader_3 += remaining_share_leader_2 / 2
+
+        # Leaders 1 and 2 gets their base share plus the distributed reduction from Leader 3
+        expected_share_leader_1 += remaining_share_leader_3 / 2
+        expected_share_leader_2 += remaining_share_leader_3 / 2
+
+        self.assertAlmostEqual(rewarder.scores["leader-1"], expected_share_leader_1, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-2"], expected_share_leader_2, places=2)
+        self.assertAlmostEqual(rewarder.scores["leader-3"], expected_share_leader_3, places=2)
+
+    def test_three_competitions_three_leaders_all_same_reductions(self):
+        """Test case 6: All competitors have the same amount of reduced shares"""
+        competitions_leaders = {
```

Comment:
so the result should be 1/3 split like without reductions, I don't know what you calculate there